npc_simulator/src/node.cpp

```
@@ -26,63 +30,92 @@ typedef bg::model::d2::point_xy<double> Point;
 typedef bg::model::linestring<Point> Line;
 typedef bg::model::polygon<Point> Polygon;
 
-NPCSimulatorNode::NPCSimulatorNode() : nh_(), pnh_("~"), tf_listener_(tf_buffer_)
+NPCSimulatorNode::NPCSimulatorNode()
+: rclcpp::Node("npc_simulator"),
+  tf_buffer_(get_clock()),
+  tf_listener_(tf_buffer_),
+  vehicle_info_(vehicle_info_util::VehicleInfo::create(*this))
 {
+  using std::placeholders::_1;
+  using std::placeholders::_2;
+
   // get parameter
-  pnh_.param<bool>("initial_engage_state", engage_state_, true);
+  engage_state_ = declare_parameter<bool>("initial_engage_state", true);
 
   // get vehicle parameter
-  vehicle_width_ = waitForParam<double>(pnh_, "/vehicle_info/vehicle_width");
-  vehicle_length_ = waitForParam<double>(pnh_, "/vehicle_info/vehicle_length");
-  vehicle_rear_overhang_ = waitForParam<double>(pnh_, "/vehicle_info/rear_overhang");
-  vehicle_base2center_ = vehicle_length_ / 2.0 - vehicle_rear_overhang_;
-
-  dummy_perception_object_pub_ =
-    pnh_.advertise<dummy_perception_publisher::Object>("output/dynamic_object_info", 10, true);
-  debug_object_pub_ = pnh_.advertise<autoware_perception_msgs::DynamicObjectArray>(
-    "output/debug_object_info", 10, true);
+  double wheel_tread{}, left_overhang{}, right_overhang{};
+  assert(get_parameter("wheel_tread", wheel_tread));
```

<p><del> I think you can instead use `double wheel_tread{declare_parameter<double>("wheel_tread").get()};` instead, which will throw an exception if the user did not provide `wheel_tread` parameter.</del></p>

Though I am a bit biased against runtime assertions, let me know if you think otherwise


Edit: I have just noticed that `VehicleInfo` was used for declaring the parameters. The [usage](https://github.com/tier4/Pilot.Auto/blob/ros2/vehicle/util/vehicle_info_util/src/vehicle_info.cpp#L51) of `declare_parameter` actually makes these `assert`s unreachable in case of a missing parameter because, the usage is same as what i described which should throw an exception on a missing parameter.

So in this case, you can just use `get_parameter` and always expect to read a parameter.

---

@cvasfi Fixed now when I use vehicle_info properly as @nnmm showed me. I was also not too sure about the use of `assert`. Now I don't even need to `declare_parameter`, vehicle info does that for me